packages/core/utils/src/product/csv-normalizer.ts

```diff
@@ -0,0 +1,520 @@
+import {
+  isPresent,
+  tryConvertToNumber,
+  tryConvertToBoolean,
+  MedusaError,
+} from "../common"
+import { AdminCreateProduct, AdminCreateProductVariant } from "@medusajs/types"
+
+/**
+ * Column processor is a function that process the CSV column
+ * and writes its value to the output
+ */
+type ColumnProcessor<Output> = (
+  csvRow: Record<string, string | boolean | number>,
+  rowColumns: string[],
+  rowNumber: number,
+  output: Output
+) => void
+
+/**
+ * Creates an error with the CSV row number
+ */
+function createError(rowNumber: number, message: string) {
+  return new MedusaError(
+    MedusaError.Types.INVALID_DATA,
+    `Row ${rowNumber}: ${message}`
+  )
+}
+
+/**
+ * Normalizes a CSV value by removing the leading "\r" from the
+ * value.
+ */
+function normalizeValue<T>(value: T): T {
+  if (typeof value === "string") {
+    return value.replace(/\\r$/, "") as T
+  }
+  return value
+}
+
+/**
+ * Parses different patterns to extract variant price iso
+ * and the region name. The iso is converted to lowercase
+ */
+function parseVariantPriceColumn(columnName: string, rowNumber: number) {
+  const priceTokens = normalizeValue(columnName)
+    .replace("Variant Price ", "")
+    .split(" ")
+
+  /**
+   * The price was specified as "Variant Price EUR" or "Variant Price USD"
+   */
+  if (priceTokens.length === 1) {
+    return {
+      region: null,
+      iso: priceTokens[0].toLowerCase(),
+    }
+  }
+
+  /**
+   * The price was specified as "Variant Price [EUR]" or "Variant Price Europe [EUR]"
```

Comment:
question: Is this description accurate? Wouldn't this be:
```
"Variant Price Europe EUR" or "Variant Price North America USD"
```

Do we need the square brackets?

Reply:
We could say the currency should always be at the end of the column

Reply:
Ah yes, now I remember. 

To consistently capture the region name, we decided to use the brackets to surround it. 

With that in mind, the current parsing is actually slightly incorrect. Here are a few examples:
```
Variant Price [Europe] EUR
Variant Price [North America] USD
Variant Price [A super duper long region name] EUR
```

Having said that, I am also fine going with the current approach. This small change would just unify how current is parsed, i.e. always as the last word and without brackets.

Reply:
Alright. So the region name is inside the brackets and the code is at last word without any brackets? If so, I will fix the logic

Reply:
Done